src/ui/common/src/components/workflows/version_selector.tsx

```diff
@@ -48,7 +58,7 @@ export const VersionSelector: React.FC = () => {
 
       let defaultBackground, hoverBackground, selectedBackground;
 
-      switch (r.status) {
+      switch (r.exec_state.status) {
```

Comment:
do we need to add null checking for exec_state here? I'm not entirely sure just by looking at the code.

Reply:
Confirmed with Saurav that this is correct.